### DISCUSSION TITLE
SpotX-Bash for SteamDeck

### DISCUSSION THREAD

**Discussion by KageSureiya29** [SpotX-Official/SpotX-Bash#11] [Q&A] [answered]
Hello. I would like to ask how to install SpotX-Bash in SteamDeck.
Recently, I am unable to install it after an update pushed by Steam.

**Comment by jetfir3** [comment]
Not sure how many SteamDeck users we have but it would be helpful to provide as much info as possible. 
Which SteamDeck update?
Which Spotify version?
Which SpotX-Bash version?
Which SXB command was used?
What was the output of SXB?
Any additional info to add?

**Comment by KageSureiya29** [comment]
sorry for the late response
~ Which SteamDeck update?
Steam Version:  1684535786
Steam Client Build Date:  Sat, May 20 06:00 UTC -08:00
Steam Web Build Date:  Sat, May 20 06:13 UTC -08:00
Steam API Version:  SteamClient020

Operating System: SteamOS 3.4.8
KDE Plasma Version: 5.26.1
KDE Frameworks Version: 5.99.0
Qt Version: 5.15.6
Kernel Version: 5.13.0-valve36-1-neptune (64-bit)


~ Which Spotify version?
1.2.11.916.geb595a67 released on 5/15/23

~ Which SpotX-Bash version?
recent release

~ Which SXB command was used?
default - used the default syntax from the documentation

~ What was the output of SXB?
Warning: Spotify not found in PATH. Searching for Spotify directory...

Error: Spotify directory not found.
Set directory path with '-P' flag. Exiting...


~ Any additional info to add?
Spotify is located at 
/home/deck/.var/app/com.spotify.Client/

**Comment by KageSureiya29** [comment]
By the way, adding new info about where Spotify install in Steam Deck
There are two directories
/home/deck/.local/share/flatpak/app/com.spotify.Client/
/home/deck/.var/app/com.spotify.Client/

I have checked it on the usr/lib directory but spotify didnt install there.

**Comment by jetfir3** [comment, marked as answer]
The "Spotify directory not found" issue mentioned in the FAQ doesn't help at all?
https://github.com/jetfir3/SpotX-Bash/wiki/SpotX%E2%80%90Bash-FAQ#why-does-spotx-bash-say-spotify-not-found-when-spotify-is-installed

From your most recent post it looks like Spotify installation via flatpak is happening (or has happened at some point) so the -P flag pointing to the Spotify directory should solve the issue (note the flatpak path/directory example in the FAQ). 

More recent builds of SpotX-Bash should now be able to find default locations when flatpak, AUR, APT and other package managers are used. If Spotify is installed then it should hopefully be automatically found... and for the cases where it isn't, you can always use the -P argument followed by the path pointing to the spotify app directory.
